### DISCUSSION TITLE
LobeChat File Ops Support

### DISCUSSION THREAD

**Discussion by AxDSan** [lobehub/lobehub#3065] [Development Question] [answered]
Hey there this discussion in regards the file support for LobeChat in general, are we going to implement RAG? what is the current roadmap for this? are there any previous discussions?

**Comment by arvinxx** [comment, marked as answer]
https://github.com/lobehub/lobe-chat/discussions/3023

**Comment by arvinxx** [comment]
Now we have supported RAG~

**Reply by AxDSan** [reply]
> Now we have supported RAG~

Just noticed! Thank you so much for this... Question is this going to be leveraged only as a paid feature?

**Reply by arvinxx** [reply]
no, we open source the whole code. So you can self-host it with vercel/docker/zeabur for free.
